### DISCUSSION TITLE
Alpine Expression Error

### DISCUSSION THREAD

**Discussion by Martin-1182** [alpinejs/alpine#3027] [1. Help]
>Hello everyone, I need a reset() method to clear the detepicker. I get an error `Uncaught ReferenceError: reset is not defined` **Can you advise me what I'm doing wrong?** THX

```
@props(['name' => '', 'placeholder' => ''])
<input x-data="datepicker(@js($getConfig()))" x-ref="datePicker" type="text" x-modelable="value" x-bind:id="$id('input')"
    placeholder="{{ $placeholder ? $placeholder : 'DD-MM-YYYY' }}" name="{{ $name }}" {{ $attributes }}
    class="mt-1 focus:ring-indigo-500 focus:border-indigo-500 block w-full shadow-sm sm:text-sm border-gray-300 rounded-md">

    <span
        x-on:click="reset()"
        class="cursor-pointer text-indigo-500 hover:text-indigo-600"
    >reset
</span>

@once
    <script>
        document.addEventListener("alpine:init", () => {
            Alpine.data('datepicker', (config) => ({
                value: null,
                init() {
                    flatpickr(this.$refs.datePicker, config).config.onChange.push(
                        function(selectedDates, dateStr, instance) {
                            this.value = dateStr;
                        }
                    );
                },
                reset() {
                    console.log('reset');
                }
            }));
        });
    </script>
@endonce


```

**Comment by SimoTod** [comment]
Your datepicket component (which defines reset) is only the input tag. Yiu are invoking that function from outside the component where is not defined

**Reply by Martin-1182** [reply]
Thank you for your response. So how could I call the reset method with the outside component in this case?

**Reply by SimoTod** [reply]
You can move x-data to the parent tag

**Reply by Martin-1182** [reply]
@SimoTod That's why I'm asking because when I move x-data to the parent tag, the datepicker stops working. I have no error in the console.

**Reply by SimoTod** [reply]
@Martin-1182 It seems to work correctly. Here is a simplified version: https://codepen.io/SimoTod/pen/ZExOamE?editors=1111

In your example, you also have an x-modelable which needs to be moved to the component

**Reply by Martin-1182** [reply]
@SimoTod  I don't know, but it doesn't work for me... After the first date selection, the datepicker dies. Your example works I looked. I'm doing something wrong...

 ```  
<div
   x-data="datepicker(@js($getConfig()))"
   x-modelable="value" 
   class="relative"
   >
    <input
        x-ref="datePicker"
        type="text"
        x-bind:id="$id('input')"
        placeholder="{{ $placeholder ? $placeholder : 'DD-MM-YYYY' }}"
        name="{{ $name }}" {{ $attributes }}
        class="mt-1 focus:ring-indigo-500 focus:border-indigo-500 block w-full shadow-sm sm:text-sm border-gray-300 rounded-md"
    >
    <span x-on:click="reset()">reset</span>
</div> 

 ```



![screen](https://agritours.invelity.com/storage/screen.png)

**Reply by SimoTod** [reply]
Try without x-modelable, I suspect you have a bug there

**Reply by Martin-1182** [reply]
@SimoTod 
I tried it anyway and it didn't help either.

**Reply by SimoTod** [reply]
Can you recreate the bug on Laravel playground and share the link?

**Reply by Martin-1182** [reply]
@SimoTod  
I won't be able to replicate it exactly, the date picker is a component that I insert into the page in several places
